air/src/air/transition/mod.rs

```diff
@@ -171,3 +172,137 @@ impl<E: FieldElement> TransitionConstraints<E> {
         result / z
     }
 }
+
+// LAGRANGE KERNEL TRANSITION CONSTRAINTS INFO
+// ================================================================================================
+
+/// Represents the transition constraints for the Lagrange kernel column, as well as the random
+/// coefficients used to linearly combine all the constraints.
+///
+/// There are `log(trace_len)` constraints, each with its own divisor, as described in
+/// [this issue](https://github.com/facebook/winterfell/issues/240).
+pub struct LagrangeKernelTransitionConstraints<E: FieldElement> {
+    lagrange_constraint_coefficients: Vec<E>,
+    divisors: Vec<ConstraintDivisor<E::BaseField>>,
+}
```

Comment:
I would move this to `src/air/lagrange/transition.rs` module.

Also, I'm not sure we need to use `ConstraintDivisor` structs here. The divisors for Lagrange kernel constraints are pretty simple and I think it might be simpler (and probably more efficient) to handle them in a specialized way here.